gammapy/cube/fit.py

```diff
@@ -172,6 +178,113 @@ def likelihood(self):
 
         return stat
 
+    def to_hdulist(self):
+        """Convert map dataset to list of HDUs.
+
+        Returns
+        -------
+        hdulist : `~astropy.io.fits.HDUList`
+            Map dataset list of HDUs.
+        """
+        # TODO: what todo about the model and background model parameters?
+        exclude_primary = slice(1, None)
+
+        hdu_primary = fits.PrimaryHDU()
+        hdulist = fits.HDUList([hdu_primary])
+        hdulist += self.counts.to_hdulist(hdu="counts")[exclude_primary]
+        hdulist += self.exposure.to_hdulist(hdu="exposure")[exclude_primary]
+        hdulist += self.background_model.map.to_hdulist(hdu="background")[exclude_primary]
+
+        if self.edisp is not None:
+            hdus = self.edisp.to_hdulist()
+            hdus["MATRIX"].name = "edisp"
+            hdus["EBOUNDS"].name = "edisp_ebounds"
+            hdulist.append(hdus["EDISP"])
+            hdulist.append(hdus["EDISP_EBOUNDS"])
+
+        if self.psf is not None:
+            hdulist += self.psf.psf_kernel_map.to_hdulist(hdu="psf")[exclude_primary]
+
+        if self.mask_safe is not None:
+            mask_safe_map = Map.from_geom(self.counts.geom, data=self.mask_safe.astype(int))
+            hdulist += mask_safe_map.to_hdulist(hdu="mask-safe")[exclude_primary]
+
+        if self.mask_fit is not None:
```

Comment:
This I wasn't 100% sure either but eventually decided to add this information to the FITS file.
I guess there are 2 options:

1. Use the single FITS file serialisation only for data-reduction and only store the data associated with this.
2. Aim for a full `MapDataset` serialisation in a single FITS file (maybe with a separate model file) and really serialise all the information.

I have a slight preference for the 2nd option. So try to have a full `MapDataset` serialisation, so that the state of an analysis can be stored / re-stored anytime. And this would include the fit mask as well. Maybe we can discuss this again.